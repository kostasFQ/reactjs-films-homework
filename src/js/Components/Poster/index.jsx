import React from 'react';
import PropTypes from 'prop-types';
import styles from './poster.scss';

const Poster = (props) => {
  const { pict } = props;
  return (
    <div>
      <div className={styles.curtian}>
        <img src={pict} className={styles.poster} alt="no poster" />
      </div>
    </div>
  );
};
export default Poster;

Poster.propTypes = {
  pict: PropTypes.string.isRequired,
};
