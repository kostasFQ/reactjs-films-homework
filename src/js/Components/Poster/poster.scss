.poster {
  display: flex;
  height: auto;
  width: 100%;
  position: relative;
  z-index: -1;
}

.curtian {
  min-width: 320px;
  background-image: linear-gradient(rgba(0, 0, 0, 0.7) 0%, rgba(255, 255, 255, 0) 30%, rgba(255, 255, 255, 0) 60%, rgba(0, 0, 0, 0.7) 100%,);
}
