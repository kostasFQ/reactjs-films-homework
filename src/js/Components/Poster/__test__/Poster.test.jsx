import React from 'react';
import renderer from 'react-test-renderer';
import Poster from '../index';
import pict from '../../../../imgs/cover-image.png';

test('test component', () => {
  const render = renderer.create(
    <Poster pict={pict} />,
  );

  const component = render.toJSON();

  expect(component).toMatchSnapshot();
  expect(component.type).toBe('img');
});
