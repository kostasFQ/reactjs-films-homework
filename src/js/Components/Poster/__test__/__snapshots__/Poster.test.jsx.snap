// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test POSTER component 1`] = `
<div>
  <div
    className="curtian"
  >
    <img
      alt="no poster"
      className="poster"
      src="test-file-stub"
    />
  </div>
</div>
`;
